01-ruby-fundamentals/raising-exceptions.md

```diff
@@ -107,11 +168,24 @@ unless order_items.method_defined? :sum && number_of_people > 0
 end
 ```
 
-This prevents your methods from being called with an invalid number people when trying to divide the check.
+This prevents our beautiful methods from being called with an invalid number of people when trying to divide the check.
+
+### Why Raise Errors Ourselves?
+
+Software development is normally a team sport.  In your career you will often write methods which will be used by other developers over the course of years.  In addition you may use a method you wrote **years** after you originally wrote it.  In such circumstances it can be easy to accidentally call a method with an invalid context.  By providing checks to ensure proper context, your methods are defensive, preventing misuse.  This can prevent more subtle and harder to debug applications.  In turn this can make your methods easier to use.
+
+We may discover more and more patterns to use `raise`. Right now, know that it's an option that will help us out in the future.
+
+## What to Look Forward To
+
+Later in the curriculum we will learn:
+  - How to write Ruby code that indicates what to do in case there's an exception (and not stop the program entirely)
+  - How to create our own types of custom exceptions, to give us better detail and context for different problems
+
 
-### Why use ArgumentError?
+## Warnings are Not Exceptions
 
-Software development is normally a team sport.  In your career you will often write methods which will be used by other developers over the course of years.  In addition you may use a method you wrote **years** after you originally wrote it.  In such circumstances it can be easy to accidentally call a method with an invalid argument.  By providing checks to ensure proper arguments, your methods are defensive, preventing misuse.  This can prevent more subtle and harder to debug applications.  In turn this can make your methods easier to use.
+Now that we have more to read in our Terminal, it's good to note that Warnings are not Exceptions or Errors. If you see a "Warning" in your Terminal but your code still runs, it's not the same as an Error or Exception halting your code.
```

Comment:
```suggestion
Now that we have more to read in our Terminal, it's good to note that Warnings are not Exceptions or Errors. If you see a "Warning" in your Terminal but your code still runs, it's not the same as an Error or Exception halting your code.

Just because a Warning doesn't stop your program from running doesn't mean we can just ignore them! Warnings are a way for the people who make Ruby to let you know that you're doing something dangerous, or confusing, or to try and tell you that your code might be doing something that you didn't intend!
```